spaces.go

```diff
@@ -0,0 +1,413 @@
+	start := ParseJSONDate(event.Start)
+	return start.Before(t)
+}
+
+// EventAfter tests if any part of an Event falls after a given time.Time. It is NOT an EventPredicate.
+func EventAfter(event Event, t time.Time) bool {
+	end := ParseJSONDate(event.End)
+	return end.After(t)
+}
+
+// EventComingWeek is an EventPredicate testing if an Event overlaps with the time period of (now, now + 7 days)
+func EventComingWeek(event Event) bool {
+	now := time.Now()
+	week := now.AddDate(0, 0, 7)
+	return EventAfter(event, now) && EventBefore(event, week)
+}
+
+// EventBetweenDays returns an EventPredicate testing if an event falls between the start and end date, inclusive.
+// It is meant to be called with start, end time.Time obtained from ParseDDMMYYDate, which by default has time = 12MN.
+func EventBetweenDays(start, end time.Time) func(Event) bool {
+	correctedEnd := end.AddDate(0, 0, 1) // Standard is 12AM, so 1 day is added to capture events on the last day
+	return func(event Event) bool {
+		return EventAfter(event, start) && EventBefore(event, correctedEnd)
+	}
+}
+
+// BookingsTodayMessage returns a string describing all bookings today.
+func (spaces Spaces) BookingsTodayMessage() string {
+	message := fmt.Sprintf("Displaying bookings for today:\n\n")
+	message += spaces.Restrict(EventToday).ToString()
+	return message
+}
+
+// BookingsNowMessage returns a string describing all bookings active this moment.
+func (spaces Spaces) BookingsNowMessage() string {
+	message := fmt.Sprintf("Displaying bookings ongoing right now (%s):\n\n", FormatTimeDate(time.Now()))
+	message += spaces.Restrict(EventNow).ToString()
+	return message
+}
+
+// BookingsComingWeekMessage returns a string describing bookings within 7 days.
+func (spaces Spaces) BookingsComingWeekMessage() string {
+	message := "Displaying bookings within 7 days from now:\n\n"
+	message += spaces.Restrict(EventComingWeek).ToString()
+	return message
+}
+
+// BookingsOnDate returns a string describing bookings on the same day as the given time.Time.
+func (spaces Spaces) BookingsOnDate(t time.Time) string {
+	message := fmt.Sprintf("Displaying bookings on %s:\n\n", FormatDate(t))
+	message += spaces.Restrict(EventOnDay(t)).ToString()
+	return message
+}
+
+// BookingsBetween returns a string describing bookings in the given interval.
+func (spaces Spaces) BookingsBetween(start, end time.Time) string {
+	message := fmt.Sprintf("Displaying bookings between %s and %s\n\n", FormatDate(start), FormatDate(end))
+	message += spaces.Restrict(EventBetweenDays(start, end)).ToString()
+	return message
+}
+
+//Spaces is the primary Cinnabot Spaces method that the end user interacts with.
+//	"/spaces" displays bookings for today.
+//	"/spaces now" displays bookings right this moment.
+//	"/spaces week" displays bookings in the next 7 days.
+//	"/spaces dd/mm/yy" displays bookings on the given date.
+//	"/spaces dd/mm/yy dd/mm/yy" displays bookings in the given interval (limited to one month++).
```

Comment:
I doubt people will use the dd/mm/yy and the date range feature much. Do you anticipate people using it?

If removing that vastly simplifies the code, I'm very much for removing it. 

We can also have a link to the calendar of events where people can find more precise details.